File: starlore-back/src/main/java/com/robin/blogback/entity/AiConfig.java
package com.robin.blogback.entity;

import com.fasterxml.jackson.annotation.JsonProperty;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("ai_configs")
public class AiConfig {
    @TableId(type = IdType.AUTO)
    private Integer id;
    private String modelKey;
    private String modelName;
    private String apiUrl;
    private String modelId;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String apiKey;
    private Boolean enabled;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: starlore-py/app/models/ai_config.py
"""AI 模型配置。"""

from datetime import datetime

from sqlalchemy import Integer, String, Boolean, DateTime
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class AiConfig(Base):
    __tablename__ = "ai_configs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    modelKey: Mapped[str] = mapped_column(String(50))  # noqa: N815
    modelName: Mapped[str] = mapped_column(String(100))  # noqa: N815
    apiUrl: Mapped[str] = mapped_column(String(500))  # noqa: N815
    modelId: Mapped[str] = mapped_column(String(100))  # noqa: N815
    apiKey: Mapped[str | None] = mapped_column(String(500))  # noqa: N815
    enabled: Mapped[bool] = mapped_column(Boolean, default=True)
    createdAt: Mapped[datetime | None] = mapped_column(DateTime)  # noqa: N815
    updatedAt: Mapped[datetime | None] = mapped_column(DateTime)  # noqa: N815
